src/components/Menu/MenuGridItem/useMenuGridItemNavContext.jsx

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from "react";
+import { NAV_DIRECTIONS } from "../../../hooks/useFullKeyboardListeners";
+import { useGridKeyboardNavigationContext } from "../../GridKeyboardNavigationContext/GridKeyboardNavigationContext";
+
+export const useMenuGridItemNavContext = ({
+  wrapperRef,
+  setActiveItemIndex,
+  getPreviousSelectableIndex,
+  getNextSelectableIndex,
+  activeItemIndex,
+  isUnderSubMenu,
+  closeMenu
+}) => {
+  /*
+   * This is an "adapter" between the Grid Keyboard Navigation mechanism and the Menu navigation mechanism.
+   * Currently, the two mechanisms work a bit differently.
+   * In the future, when the Menu component will use a GridKeyboardNavigationContext, this adapter shouldn't be needed anymore.
+   */
+  const innerKeyboardContext = useGridKeyboardNavigationContext([], wrapperRef);
+  const keyboardContext = useMemo(
+    () => ({
+      onOutboundNavigation: (elementRef, direction) => {
+        innerKeyboardContext.onOutboundNavigation(elementRef, direction);
+
+        switch (direction) {
```

Comment:
no right because of sub menus ? - i guess we don't allow sub menu and a MenuItemGrid

Reply:
if we would have supported submenus inside a `MenuGridItem`, then we should have added a handler for outbound right navigation... however, I don't think such behavior makes any sense (I mean, it will be a grid that opens a sub menu when navigating to the right of it. Sounds odd, no? 🤔 ).
In any case, if we'll migrate the Menu implementation to use the grid keyboard navigation, it will be easier to support such (odd) behaviors 😄